### PR TITLE
Add two new filters for the team stats

### PR DESCRIPTION
**Pull request by CSNWEB**
- Allow filtering by only season, excluding all events
- Allow filtering by the last three seasons
- Follows a request by Shintaro on Discord

### CODE REVIEW

**Comment by CSNWEB** [comment]
This includes formatting from phpfmt and phpcs, I can remove that again if needed, but generally I prefer to Autoformat all code.

File: plugins/rikki/loungestatistics/components/TeamStatistics.php [hunk 1]
```diff
@@ -38,56 +33,87 @@ public function init()
...
+                }
+            ) // unique does not work on these
             ->sortByDesc('created_at')->values();
 
+        $this->lastThree = $this->participatedSeasons->filter(
+            function ($season) {
+                return in_array($season->type, [1, 2]);
+            }
+        )->
+            take(3)->pluck('title', 'id');
```
[the remark this answers is not in the document]
Comment (CSNWEB) [reply plugins/rikki/loungestatistics/components/TeamStatistics.php:59]: Yes

File: plugins/rikki/loungestatistics/components/TeamStatistics.php [hunk 2]
```diff
@@ -38,56 +33,87 @@ public function init()
...
-            return $map_array['picks_by'] + $map_array['picks_vs'];
-        });
+        $this->maps = $maps->sortByDesc(
+            function ($map_array) {
+                return $map_array['picks_by'] + $map_array['picks_vs'];
+            }
+        );
     }
 
     public function onSeasonChange()
```
[the remark this answers is not in the document]
Comment (CSNWEB) [reply plugins/rikki/loungestatistics/components/TeamStatistics.php:86]: Good point. I changed TeamStatistics to always work with an array of ids.

File: plugins/rikki/loungestatistics/components/TeamStatistics.php [hunk 3]
```diff
@@ -55,29 +55,27 @@ function ($group) {
...
-            take(3)->pluck('title', 'id');
+        )->take(3)->pluck('title', 'id');
 
-        $this->selectedSeason = $this->participatedSeasons
-            ->filter(
-                function ($season) {
-                    return $season->is_active;
-                }
-            )->last() ?? $this->participatedSeasons->first();
+        $this->selectedSeasons = $this->participatedSeasons->filter(
```
Comment (Fabian-Sommer) [inline plugins/rikki/loungestatistics/components/TeamStatistics.php:60]: nit: This should get an array assigned (and can then be passed through in l 66)

**Review by Fabian-Sommer — APPROVED** [review]
LGTM % nit

File: plugins/rikki/loungestatistics/components/TeamStatistics.php (hunk 3 above)
Comment (CSNWEB) [reply plugins/rikki/loungestatistics/components/TeamStatistics.php:60]: I split up that logic a bit, adding the array logic on top of the old one was a bit convoluted.

Comment (Fabian-Sommer) [reply plugins/rikki/loungestatistics/components/TeamStatistics.php]: I think the problem there was that the non-array ended up being persisted in $this->selectedSeasons at the end of the function. Looks good now though

**Comment by eayildiz** [comment]
Razhag | Madness from discord asked the "the last three seasons" filter for players as well. Would it be okay if I try to work on that?

**Pull request merged**